feat(demo-mode) also sync ProjectDebugFile and ProguardArtifactRelease

constantinius opened the pull request:
Closes https://linear.app/getsentry/issue/TET-528/extend-demo-mode-sync-task-to-handle-debug-files-and-proguard

constantinius committed:
feat(demo-mode) also sync ProjectDebugFile and ProguardArtifactRelease

obostjancic reviewed (approved):
nit: I'd rename this task to something like `sync_debug_artifacts` and introduce 3 sub functions like:  
`_sync_artifact_bundles(source, target)`, `_sync_debug_files_(source, target)`, `_sync_proguard_files(source, target)`

Otherwise LGTM! 🚀

constantinius committed:
fix(demo-mode): using already synced project debug file and correct project id

constantinius committed:
test(demo-mode): add tests for ProjectDebugFile and ProguardArtifactRelease sync

constantinius committed:
fix(demo-mode): new name for main task: sync_debug_artifacts
Dedicated  functions for project debug files and proguard artifact releases.
Making sure that project debug files within proguard releases are always synced if not yet done

constantinius committed:
fix(demo-mode) typing issue

constantinius committed:
fix(demo-mode): mypy type issue

constantinius committed:
fix(demo-mode): name of task in CELERYBEAT_SCHEDULE
